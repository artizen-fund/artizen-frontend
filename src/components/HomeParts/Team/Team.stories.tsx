import Team from './'

export default {
  title: 'components/Team',
  component: Team,
  argTypes: {},
}

export const TeamComponent = (props: any) => <Team {...props} />
