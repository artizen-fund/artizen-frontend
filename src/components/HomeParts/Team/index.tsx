import styled from 'styled-components'
import { PagePadding } from '@components'
import { assetPath } from '@lib'
import { breakpoint, typeface, typography } from '@theme'
import { team } from '@copy/home'

const Curators = () => (
  <PagePadding>
    <Wrapper>
      <Headline>{team.title}</Headline>
      <Copy>{team.copy}</Copy>
      <ul>
        {team.curators.map((curator, i) => (
          <Curator key={`curator-${i}`}>
            <img src={assetPath(`${curator.portrait}?fm=webp`)} alt={curator.name} />
            <Description>
              <Name>{curator.name}</Name>
              <Title>{curator.title}</Title>
            </Description>
          </Curator>
        ))}
      </ul>
    </Wrapper>
  </PagePadding>
)

const Wrapper = styled.div`
  text-align: center;
  margin: auto;
  display: flex;
  flex-direction: column;
  justify-content: center;

  ul {
    margin: 50px auto auto auto;
    display: grid;
    grid-template-columns: repeat(2, 150px);
    grid-template-rows: repeat(2, 1fr);
    gap: 20px;
    @media only screen and (min-width: ${breakpoint.tablet}px) {
      grid-template-columns: repeat(4, 1fr);
      grid-template-rows: auto;
    }
  }
`

const Headline = styled.h2`
  ${typography.title.l2}
`

const Copy = styled.p`
  ${typography.body.l2}
`

const Curator = styled.div`
  img {
    max-width: 100%;
  }
  width: 150px;
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    width: 175px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    width: 222px;
  }
  overflow: hidden;
  border-radius: 24px;
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.12);
`

const Description = styled.div`
  padding: 24px 0 24px 0;
`

const Name = styled.div`
  ${typography.label.l1}
`

const Title = styled.div`
  ${typography.label.l2}
  font-family: ${typeface.mackinacPro};
  font-weight: 500;
`

export default Curators
