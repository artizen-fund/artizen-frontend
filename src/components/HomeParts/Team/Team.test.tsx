import renderer from 'react-test-renderer'
import Team from './'

describe('Team', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Team />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
