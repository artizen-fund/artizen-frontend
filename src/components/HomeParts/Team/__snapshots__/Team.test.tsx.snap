// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team renders correctly 1`] = `
<div
  className="sc-45d68d47-0 kVBvae"
>
  <div
    className="sc-7aeeab03-0 jxPjHK"
  >
    <h2
      className="sc-7aeeab03-1 cBWYFV"
    >
      Our Team
    </h2>
    <ul>
      <div
        className="sc-7aeeab03-2 jkProv"
      >
        <img
          alt="René Pinnell"
          src="/assets/illustrations/team/rene.jpg?fm=webp"
        />
        <div
          className="sc-7aeeab03-3 kRHDBJ"
        >
          <div
            className="sc-7aeeab03-4 fSMUjm"
          >
            René Pinnell
          </div>
          <div
            className="sc-7aeeab03-5 fqjUpH"
          >
            Co-Founder & CEO
          </div>
        </div>
      </div>
      <div
        className="sc-7aeeab03-2 jkProv"
      >
        <img
          alt="Ruben Campos"
          src="/assets/illustrations/team/ruben.jpg?fm=webp"
        />
        <div
          className="sc-7aeeab03-3 kRHDBJ"
        >
          <div
            className="sc-7aeeab03-4 fSMUjm"
          >
            Ruben Campos
          </div>
          <div
            className="sc-7aeeab03-5 fqjUpH"
          >
            Co-Founder & CTO
          </div>
        </div>
      </div>
      <div
        className="sc-7aeeab03-2 jkProv"
      >
        <img
          alt="Simon Corry"
          src="/assets/illustrations/team/simon.jpg?fm=webp"
        />
        <div
          className="sc-7aeeab03-3 kRHDBJ"
        >
          <div
            className="sc-7aeeab03-4 fSMUjm"
          >
            Simon Corry
          </div>
          <div
            className="sc-7aeeab03-5 fqjUpH"
          >
            Head of Product
          </div>
        </div>
      </div>
      <div
        className="sc-7aeeab03-2 jkProv"
      >
        <img
          alt="Dov Heichemer"
          src="/assets/illustrations/team/dov.jpg?fm=webp"
        />
        <div
          className="sc-7aeeab03-3 kRHDBJ"
        >
          <div
            className="sc-7aeeab03-4 fSMUjm"
          >
            Dov Heichemer
          </div>
          <div
            className="sc-7aeeab03-5 fqjUpH"
          >
            Head of Community
          </div>
        </div>
      </div>
      <div
        className="sc-7aeeab03-2 jkProv"
      >
        <img
          alt="Juju Parchimowicz"
          src="/assets/illustrations/team/juju.jpg?fm=webp"
        />
        <div
          className="sc-7aeeab03-3 kRHDBJ"
        >
          <div
            className="sc-7aeeab03-4 fSMUjm"
          >
            Juju Parchimowicz
          </div>
          <div
            className="sc-7aeeab03-5 fqjUpH"
          >
            Product Design & Art Direction
          </div>
        </div>
      </div>
      <div
        className="sc-7aeeab03-2 jkProv"
      >
        <img
          alt="Eric Katerman"
          src="/assets/illustrations/team/eric-k.jpg?fm=webp"
        />
        <div
          className="sc-7aeeab03-3 kRHDBJ"
        >
          <div
            className="sc-7aeeab03-4 fSMUjm"
          >
            Eric Katerman
          </div>
          <div
            className="sc-7aeeab03-5 fqjUpH"
          >
            Mathematician + Blockchain Dev
          </div>
        </div>
      </div>
      <div
        className="sc-7aeeab03-2 jkProv"
      >
        <img
          alt="Zsofie Tubel"
          src="/assets/illustrations/team/zsofia.jpg?fm=webp"
        />
        <div
          className="sc-7aeeab03-3 kRHDBJ"
        >
          <div
            className="sc-7aeeab03-4 fSMUjm"
          >
            Zsofie Tubel
          </div>
          <div
            className="sc-7aeeab03-5 fqjUpH"
          >
            Blockchain Dev
          </div>
        </div>
      </div>
      <div
        className="sc-7aeeab03-2 jkProv"
      >
        <img
          alt="Eric Jacobsen"
          src="/assets/illustrations/team/eric-j.jpg?fm=webp"
        />
        <div
          className="sc-7aeeab03-3 kRHDBJ"
        >
          <div
            className="sc-7aeeab03-4 fSMUjm"
          >
            Eric Jacobsen
          </div>
          <div
            className="sc-7aeeab03-5 fqjUpH"
          >
            Frontend Dev
          </div>
        </div>
      </div>
    </ul>
  </div>
</div>
`;
